src/analysisd/alerts/exec.c

```diff
@@ -168,3 +170,110 @@ void OS_Exec(int execq, int *arq, const Eventinfo *lf, const active_response *ar
 
     return;
 }
+
+/**
+ * @brief get the IP.
+ *
+ * @param[in] lf Event information.
+ * @return const char* on success or NULL on failure.
+ */
+const char* get_ip(const Eventinfo *lf)
+{
+    const char *ip;
+    ip = "-";
+
+    if (strncmp(lf->srcip, "::ffff:", 7) == 0) {
+        ip = lf->srcip + 7;
+    } else {
+        ip = lf->srcip;
+    }
+
+    /* Check if IP is to be ignored */
+    if (Config.white_list) {
+        if (OS_IPFoundList(ip, Config.white_list)) {
+            return NULL;
+        }
+    }
+
+    /* Check if it is a hostname */
+    if (Config.hostname_white_list) {
+        size_t srcip_size;
+        OSMatch **wl;
+
+        srcip_size = strlen(ip);
+
+        wl = Config.hostname_white_list;
+        while (*wl) {
+            if (OSMatch_Execute(ip, srcip_size, *wl)) {
+                return NULL;
+            }
+            wl++;
+        }
+    }
+    return ip;
+}
+        
+/**
+ * @brief Build the string message
+ *
+ * @param[in] lf Event information.
+ * @param[in] ar Active Response information.
+ * @param[in] ip IP to send the specific message.
+ * @param[in] user
+ * @param[in] filename
+ * @param[in] extra_args Added arguments.
+ * @param[out] temp_msg Message in string format.
+ * @return void.
+ */
+void getActiveResponseInString( const Eventinfo *lf, 
+                                const active_response *ar, 
+                                const char *ip, 
+                                const char *user, 
+                                char *filename,
+                                char *extra_args, 
+                                char *temp_msg)
+{
+    snprintf(temp_msg, OS_SIZE_1024,
+            "%s %s %s %ld.%ld %d %s %s %s",
+            ar->name,
+            user,
+            ip,
+            (long int)lf->time.tv_sec,
+            __crt_ftell,
+            lf->generated_rule->sigid,
+            lf->location,
+            filename ? filename : "-",
+            extra_args ? extra_args : "-");
+}
+
+/**
+ * @brief Add the header to the message to send to remoted
+ *
+ * @param[in] lf Event information.
+ * @param[in] ar Active Response information.
+ * @param[in] msg Message that can be in JSON or string format
+ * @param[out] exec_msg Complete massage containing the message and the header.
+ * @return void.
+ */
+void get_exec_msg(const Eventinfo *lf, const active_response *ar, const char *msg, char *exec_msg)
+{
+    char temp_msg[OS_SIZE_1024 + 1];
```

Comment:
This size is too short. Use at least 4096 instead.

Reply:
But in this case, those 1024 are only for the header, which is composed of 3 characters and 2 strings.